glean-core/android/src/main/java/mozilla/telemetry/glean/private/StringMetricType.kt

```diff
@@ -23,17 +23,23 @@ import mozilla.telemetry.glean.rust.toBoolean
  *
  * The string API only exposes the [set] method, which takes care of validating the input
  * data and making sure that limits are enforced.
+ *
+ * The internal constructor is only used by labeled metrics directly.
```

Comment:
nit: you can change this to `"The internal constructor is only used by [LabeledMetricType] directly"` so that the generated docs point to that type.